tools/python/register_custom_ops_pytorch_exporter.py

```diff
@@ -16,6 +16,25 @@ def register_custom_op():
     """
 
     # Symbolic definition
+    def grid_sample(g, self, input, mode, padding_mode, align_corners):
+        # mode
+        #   'bilinear'      : onnx::Constant[value={0}]
+        #   'nearest'       : onnx::Constant[value={1}]
+        #   'bicubic'       : onnx::Constant[value={2}]
+        # padding_mode
+        #   'zeros'         : onnx::Constant[value={0}]
+        #   'border'        : onnx::Constant[value={1}]
+        #   'reflection'    : onnx::Constant[value={2}]
+        mode = sym_help._maybe_get_const(mode, "i")
+        padding_mode = sym_help._maybe_get_const(padding_mode, "i")
+        mode_str = ['bilinear', 'nearest', 'bicubic'][mode]
+        padding_mode_str = ['zeros', 'border', 'reflection'][padding_mode]
+        align_corners = int(sym_help._maybe_get_const(align_corners, "b"))
+        return g.op("com.microsoft::GridSample", self, input,
+                    mode_s=mode_str,
+                    padding_mode_s=padding_mode_str,
+                    align_corners_i=align_corners)
```

Comment:
Exporter is unable to perform standard onnx shape inference in custom symbolic function as contrib ops. Models with dynamic shape inputs are affected. 
So It may better to add `.setType(self.type())` the end of the g.op().

Reply:
@XiyinOSS Thanks for our team @BowenBao reminding. The setType() function will also assign the shape of the output, So we need add the shape with with_sizes() too. such as using the below code:

    from torch.onnx.symbolic_helper import _get_tensor_sizes
    # (N, C, H, W) (N, H_out, W_out, 2) => (N, C, H_out, W_out)
    output_shape = _get_tensor_sizes(self)[:2] + _get_tensor_sizes(input)[1:3]
    return g.op("com.microsoft::GridSample", self, input,
                mode_s=mode_str,
                padding_mode_s=padding_mode_str,
                align_corners_i=align_corners).setType(self.type().with_sizes(output_shape))

Reply:
I am a bit confused here. Seems even without setType it works already. Do we really need add the type/dimension explicitly?

Reply:
It seems that is not work for previous version of PyTorch. We may just remove it and add some comments.